systemtests/src/main/java/com/github/streamshub/systemtests/resources/NamespaceManager.java

```diff
@@ -0,0 +1,151 @@
+package com.github.streamshub.systemtests.resources;
+
+import com.github.streamshub.systemtests.logs.CollectorElement;
+import io.fabric8.kubernetes.api.model.NamespaceStatus;
+import io.skodjob.testframe.resources.KubeResourceManager;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class NamespaceManager {
```

Comment:
Could you please elaborate a little bit more why is this needed for Console tests?

Reply:
The NamespaceManager ensures proper isolation and cleanup of namespaces during Console tests. It prevents namespace leaks, maintains test suite consistency, and allows efficient tracking of created namespaces and logcollection. Without it, tests could interfere with each other, leading to flaky results.

Reply:
Is TF collecting logs automatically or do i need to collect them like this? If so, then how do i get the access to namespaces from the TF? From my understanding of TF `KubeResourceManager.STORED_RESOURCES` is the handle you talk about, right?

`
List<String> namespaces = NamespaceManager.getInstance().getListOfNamespacesForTestClassAndTestCase(testClass, testCase);
        if (namespaces.isEmpty()) {
            LOGGER.warn("This test did not have any namespaces");
        } else {
            testCaseCollector.collectFromNamespaces(namespaces.toArray(new String[0]));
        }
`

Reply:
Also am i missing some info, or is TF capable of preparing  the namespace with pull secrets? if not, what would be the correct approach? I think i would still require to handle these situations, or not?

Reply:
Namespace manager was originally used in Strimzi to handle properly parallel test execution, but even this can be now handled by TF. In case you will find some place for improvement related to this, we can discuss it and implement if needed. but so far it should be in place for most of the usecases.

Reply:
Cool, I will use that this was then, thanks.